src/Analyzers/CSharp/Analyzers/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationDiagnosticAnalyzer.cs

```diff
@@ -93,18 +98,61 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
                     _ => null,
                 };
             }
+            else if (objectCreation.Parent.IsKind(SyntaxKind.ArrayInitializerExpression) &&
+                objectCreation.Parent.Parent is ArrayCreationExpressionSyntax arrayCreation)
+            {
+                typeNode = arrayCreation.Type.ElementType;
+            }
+            else if (objectCreation.Parent.IsKind(SyntaxKind.CollectionInitializerExpression) &&
+                objectCreation.Parent.Parent is ObjectCreationExpressionSyntax collectionObjectCreation)
+            {
+                var collectionTypeSymbol = semanticModel.GetTypeInfo(collectionObjectCreation, cancellationToken).Type!;
+                var targetTypeSymbol = semanticModel.GetTypeInfo(objectCreation, cancellationToken).ConvertedType!;
```

Comment:
I'm not much comfortable with null suppressions. But will let others weigh on that.

Reply:
This was temporary while I was making tests pass, forgot to take a second look at it. Fixed now.